core/src/apps/webauthn/__init__.py

```diff
@@ -749,3 +1230,413 @@ def msg_error(cid: int, code: int) -> Cmd:
 
 def cmd_error(cid: int, code: int) -> Cmd:
     return Cmd(cid, _CMD_ERROR, ustruct.pack(">B", code))
+
+
+def cbor_error(cid: int, code: int) -> Cmd:
+    return Cmd(cid, _CMD_CBOR, ustruct.pack(">B", code))
+
+
+def cbor_make_credential(req: Cmd, dialog_mgr: DialogManager) -> Optional[Cmd]:
+    if not storage.is_initialized():
+        if __debug__:
+            log.warning(__name__, "not initialized")
+        return cbor_error(req.cid, _ERR_OPERATION_DENIED)
+
+    try:
+        param = cbor.decode(req.data[1:])
+        rp_id = param[_MAKECRED_CMD_RP]["id"]
+        rp_id_hash = hashlib.sha256(rp_id).digest()
+
```

Comment:
According to the Webauthn you should check 'if all the supplied parameters are syntactically well-formed and of the correct length' right here, CTAP2 doesn't say nothing about the position of the check

Reply:
This depends on how broadly we interpret the term "syntactically well-formed". If it's limited in the narrow sense to CBOR encoding, then we are perfectly OK. If interpreted broadly, then in case of multiple errors, we might handle some cases in the wrong order. Unless some test tool explicitly checks this, I really don't think that getting the error codes exactly right in case of multiple errors is that much of a big deal, if it means messing-up the code. As far as I can tell, from a security perspective or a user perspective it is irrelevant.